subsys/bluetooth/mesh/adv.c

```diff
@@ -99,7 +100,15 @@ struct net_buf *bt_mesh_adv_create(enum bt_mesh_adv_type type,
 				   enum bt_mesh_adv_tag tag,
 				   uint8_t xmit, k_timeout_t timeout)
 {
-	return bt_mesh_adv_create_from_pool(&adv_buf_pool, adv_alloc, type,
+#if defined(CONFIG_BT_MESH_RELAY)
+	if (tag & BT_MESH_RELAY_ADV) {
```

Comment:
Probably this is more according to the Zephyr paradigm but up to you
```suggestion
	if ((tag & BT_MESH_RELAY_ADV) && IS_ENABLED(CONFIG_BT_MESH_RELAY)) {
```

Reply:
This doesn't work because of `NET_BUF_POOL_DEFINE` implementation. The pool will stay in the binary even if `BT_MESH_RELAY` set to `n`. Therefore I used `ifdef`.